src/file/format/ini.rs

```diff
@@ -3,35 +3,45 @@ use std::error::Error;
 use ini::Ini;
 
 use crate::map::Map;
-use crate::value::{Value, ValueKind};
+use crate::value::{Table, Value, ValueKind};
 
 pub fn parse(
     uri: Option<&String>,
     text: &str,
 ) -> Result<Map<String, Value>, Box<dyn Error + Send + Sync>> {
-    let mut map: Map<String, Value> = Map::new();
-    let i = Ini::load_from_str(text)?;
-    for (sec, prop) in i.iter() {
-        match sec {
-            Some(sec) => {
-                let mut sec_map: Map<String, Value> = Map::new();
-                for (k, v) in prop.iter() {
-                    sec_map.insert(
-                        k.to_owned(),
-                        Value::new(uri, ValueKind::String(v.to_owned())),
-                    );
-                }
-                map.insert(sec.to_owned(), Value::new(uri, ValueKind::Table(sec_map)));
-            }
-            None => {
-                for (k, v) in prop.iter() {
-                    map.insert(
-                        k.to_owned(),
-                        Value::new(uri, ValueKind::String(v.to_owned())),
-                    );
-                }
-            }
-        }
+    let value = from_ini(uri, Ini::load_from_str(text)?);
+
+    match value.kind {
+        ValueKind::Table(map) => Ok(map),
+
+        _ => Ok(Map::new()),
+    }
+}
+
+fn from_ini(
+    uri: Option<&String>,
+    data: Ini,
+) -> Value {
+    let mut map = Map::new();
+
+    let mut sections: Map<Option<&str>, Table> = data.into_iter().map(|(section, props)| {(
+        section,
+        props.iter().map(|(k, v)| {(
+            k.to_owned(),
+            Value::new(uri, ValueKind::String(v.to_owned())),
+        )}).collect()
+    )}).collect();
+
+    // These (optional) properties should exist top-level alongside sections:
+    if let Some(sectionless) = sections.remove(&None) {
+        map.extend(sectionless);
     }
```

Comment:
> but they're optional, so I figured a conditional was necessary?

Well, `Option` is an iterator, so calling `into_iter()` on it either returns an iterator yielding one element (in the `Some` case) or no element (in the `None` case). So I _think_ this could work.

Reply:
Tried a few attempts with your suggestion but kept getting failures like:

![image](https://github.com/mehcode/config-rs/assets/5098581/2178ce90-1750-4333-877b-8128b5d59165)

However it seems to work well if I just unwrap the option? Tests are passing with the sectionless properties and I verified that everything works fine if there is no sectionless properties too.

Reply:
Hm okay. Unwrapping is not an option if we can implement it without unwrapping.